### PR TITLE
Merge branch 'master' into kong-dev

### PR DESCRIPTION
**Pull request by gruceo**
Merge branch 'master' (in sync with upstream/master) into kong-dev.

### CODE REVIEW

File: pgmoon/init.lua [hunk 1]
```diff
@@ -964,7 +964,11 @@ do
             end
             return self.sock:sslhandshake(false, nil, self.config.ssl_verify)
           else
-            return self.sock:tlshandshake({ verify = self.config.ssl_verify, client_cert = luasec_opts.certificate, client_priv_key = luasec_opts.key })
+            if self.sock.tlshandshake then
```
Comment (gruceo) [inline pgmoon/init.lua:967]: Without this change, the tests fail with https://github.com/Kong/pgmoon/runs/7117841709?check_suite_focus=true

Comment (gruceo) [reply pgmoon/init.lua:967]: @Suika-Kong

**Comment by gruceo** [comment]
@dndx @fffonion @Suika-Kong Can you take a look at this PR please? It's syncing our dev branch with latest changes from upstream.

File: pgmoon-2.3.0-1.rockspec [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package = "pgmoon"
-version = "2.2.3-1"
+version = "2.3.0-1"
```
Comment (StarlightIbuki) [inline pgmoon-2.3.0-1.rockspec:2]: I don't have a context. So I'm not sure why we have a larger version number than [leafo/pgmoon](https://github.com/leafo/pgmoon), and we bump the version when merging.

File: pgmoon/init.lua (hunk 1 above)
Comment (StarlightIbuki) [reply pgmoon/init.lua:967]: I found that `sslhandshake()` is not from the socket module but is assigned by pgmoon. And `tlshandshake()` is not assigned anywhere, nor is it from the socket module. I need some time to recall why I wrote that in the first place.

**Review by StarlightIbuki — APPROVED** [review]
<s>I have no objection to the change. But I need to know why `tlshandshake()` was used.</s>
Premature approval.

File: pgmoon/init.lua (hunk 1 above)
Comment (StarlightIbuki) [reply pgmoon/init.lua:967]: And you need to also change the `init.moon`.

**Review by StarlightIbuki — CHANGES_REQUESTED** [review]

File: pgmoon/init.lua (hunk 1 above)
Comment (StarlightIbuki) [reply pgmoon/init.lua:967]: It seems we have removed the `tlshandshake()` function for `nginx` socket.
And what surprises me more is that `sslhandshake()` receives `opts` table for `luasocket`(what we use for init phase) but we call it with 3 arguments.

Comment (gruceo) [reply pgmoon/init.lua:967]: https://github.com/Kong/pgmoon/pull/22/commits/67dc4318721e84075b431273ceaea940016bb68e

File: pgmoon-2.3.0-1.rockspec (hunk 2 above)
Comment (gruceo) [reply pgmoon-2.3.0-1.rockspec:2]: We bump the version when merging so we can also bump our pgmoon fork on kong-ee .requirements.

**Review by tyler-ball — APPROVED** [review]

**Review by StarlightIbuki — APPROVED** [review]

**Pull request merged**